pandas/tests/frame/apply/test_frame_apply.py

```diff
@@ -1131,6 +1131,21 @@ def test_demo(self):
         )
         tm.assert_frame_equal(result.reindex_like(expected), expected)
 
+    def test_agg_with_unhashable_name_column_value(self):
+        # GH 36212 - Column name is "name"
+        data = {"name": ["foo", "bar"]}
+        df = pd.DataFrame(data)
+
+        # result's name should be None
+        result = df.agg({"name": "count"})
+        expected = pd.Series({"name": 2})
+        tm.assert_series_equal(result, expected)
+
+        # Check if name is preserved when aggregating series instead
+        result = df["name"].agg({"name": "count"})
+        expected = pd.Series({"name": 2}, name="name")
+        tm.assert_series_equal(result, expected)
+
     def test_agg_multiple_mixed_no_warning(self):
```

Comment:
Just wanna add an extra note: `df.agg()` returns a series instead of a dataframe for this particular input shape, hence the use of `assert_series_equal`